### PR TITLE
Adding Docker login to Github bulid-and-push job

### PR DESCRIPTION
**Pull request by danjbh**
## what & why?

* Adding Docker login step to build-and-push job since it's currently borked

#### https://github.com/docker/build-push-action/issues/88#issuecomment-688992823

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -14,11 +14,15 @@ jobs:
     steps:
       - name: "Checkout source code at current commit"
         uses: actions/checkout@v2
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
```
Comment (osterman) [inline .github/workflows/docker.yml]:
```suggestion
        if: github.event.pull_request.head.repo.full_name == github.repository
```

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -14,11 +14,15 @@ jobs:
...
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
       - name: "Build and push docker image to DockerHub"
         uses: docker/build-push-action@v2
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_PASSWORD }}
           repository: ${{ github.repository }}
           registry: registry-1.docker.io
           tag_with_ref: true
```
Comment (osterman) [inline .github/workflows/docker.yml]:
```suggestion
          tag_with_ref: true
          push: {{ github.event.pull_request.head.repo.full_name == github.repository }}
```

Comment (osterman) [reply .github/workflows/docker.yml]: this says `push` when it's not a fork =)

File: .github/workflows/docker.yml (hunk 1 above)
Comment (osterman) [reply .github/workflows/docker.yml]: this says, login when it's not a fork

**Review by osterman — DISMISSED** [review]

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -14,12 +14,17 @@ jobs:
...
-      - name: "Build and push docker image to DockerHub"
-        uses: docker/build-push-action@v2
+      - name: Login to DockerHub
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: "Build and push docker image to DockerHub"
+        uses: docker/build-push-action@v2
```
Comment (osterman) [inline .github/workflows/docker.yml:44]:
```suggestion
        uses: docker/build-push-action@v1
```

**Comment by danjbh** [comment]
Alright @osterman , it took a bit of tinkering to get the rest of the steps sorted out, but it's in a working state at this point, as far as I can tell at least. 

Definitely have a look at the `Prep` step in bash logic to see if you think we need any tweaks there. I tried to mirror what I think how the previous tagging was working at least.

Let me know!

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -14,12 +14,34 @@ jobs:
     steps:
       - name: "Checkout source code at current commit"
         uses: actions/checkout@v2
-      - name: "Build and push docker image to DockerHub"
-        uses: docker/build-push-action@v2
+      - name: Prepare
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        id: Prep
```
Comment (osterman) [inline .github/workflows/docker.yml]:
```suggestion
        id: prepare
```

File: .github/workflows/docker.yml [hunk 5]
```diff
@@ -14,12 +14,34 @@ jobs:
     steps:
       - name: "Checkout source code at current commit"
         uses: actions/checkout@v2
-      - name: "Build and push docker image to DockerHub"
-        uses: docker/build-push-action@v2
+      - name: Prepare
```
Comment (osterman) [inline .github/workflows/docker.yml]:
```suggestion
      - name: Prepare tags for Docker image
```

File: .github/workflows/docker.yml [hunk 6]
```diff
@@ -14,12 +14,34 @@ jobs:
...
-          repository: ${{ github.repository }}
-          registry: registry-1.docker.io
-          tag_with_ref: true
-          tag_with_sha: true
+      - name: "Build and push docker image to DockerHub"
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+          tags: ${{ steps.prep.outputs.tags }}
```
Comment (osterman) [inline .github/workflows/docker.yml]:
```suggestion
          tags: ${{ steps.prepare.outputs.tags }}
```

**Review by osterman — APPROVED** [review]

**Pull request merged**